doc/reference/annotations.rst

```diff
@@ -98,6 +98,36 @@ set the value via reflection, but you may change this to use a public method ins
         }
     }
 
+    class Shop
+    {
+        /**
+         * @AccessType("public_method", naming="camel_case")
+         * @ReadOnly
+         */
+        private $full_address;
+
+        public function getFullAddress()
+        {
+            return $this->full_address;
+        }
+    }
+
+
+To use accessors instead of properties by default, modify builder:
+
+.. code-block :: php
+
+    <?php
+    use JMS\Serializer\Accessor\Updater\ClassAccessorUpdater;
+    use JMS\Serializer\SerializerBuilder;
+
+    $serializerBuilder = SerializerBuilder::create()->build();
+    $serializerBuilder->setPropertyUpdater(new ClassAccessorUpdater(

```

Comment:
for the same reasons https://github.com/schmittjoh/JMSSerializerBundle/pull/427 was not accepted, even here will prefer not allowing to change the access type at runtime but asking explicit configurations via metadata